org.eclipse.gef/src/org/eclipse/gef/tools/AbstractTool.java

```diff
@@ -1113,30 +1136,38 @@ public void mouseMove(MouseEvent me, EditPartViewer viewer) {
 			boolean b5 = getCurrentInput().isMouseButtonDown(5);
 			getCurrentInput().verifyMouseButtons = true;
 			getCurrentInput().setInput(me);
-			if (b1)
+			if (b1) {
 				handleButtonUp(1);
-			if (b2)
+			}
+			if (b2) {
 				handleButtonUp(2);
-			if (b3)
+			}
+			if (b3) {
 				handleButtonUp(3);
-			if (b4)
+			}
+			if (b4) {
 				handleButtonUp(4);
-			if (b5)
```

Comment:
Do we really use an if-else chain for the buttons 1 to 5...?

It doesn't need to be part of this change, but perhaps it's worth adding a getMouseButton() to the Input class.

Reply:
I saw that two and thought this is a target for a next PR.